### ISSUE TITLE
feat(gcf-utils): logger should keep scope for all log statements

### ISSUE THREAD

**Issue by chingor13** [googleapis/repo-automation-bots#777]
In gcf-utils, we can capture the repository/other trigger information and add that data to all structured logs. This will make logs much more discoverable in the logging console.

**Comment by azizsonawalla** [comment]
Adding an example here of what we're looking for:

```ts
(in gcf-utils.ts)

const triggerInfo  = {                                     // this is what we parse from incoming payload
    trigger: {
        trigger_type: 'GITHUB_WEBHOOK',
        trigger_sender: 'some sender',
        payload_hash: '123456',
        trigger_source_repo: {
            owner: 'foo owner',
            owner_type: 'Org',
            repo_name: 'bar name',
            url: 'some url',
        },
    },
};

...

// bind the triggerInfo above to the exported logger
```

```ts
(in some-bot.ts)

logger.info({ some-property: 'some-value' });
```
```
(Cloud Logs Viewer)

{
    insertId: "xyz",
    jsonPayload: {
        some-property: "some-value",                // what the bot logged
        trigger: {                                   // what was automatically attached to logs
            trigger_type: 'GITHUB_WEBHOOK',
            trigger_sender: 'some sender',
            payload_hash: '123456',
            trigger_source_repo: {
                owner: 'foo owner',
                owner_type: 'Org',
                repo_name: 'bar name',
                url: 'some url',
            },
        },
    },
    resource: {
        type: "cloud-function",
        labels: {
            function_name: "some-bot",
            project_id: "repo-automation-bots",
            region: "us-central1"
        },
    },
    timestamp: "12345",
    severity: "INFO",
    labels: {
        execution_id: "abcd"
    }
}
```

**Comment by tmatsuo** [comment]
@chingor13 Why did we re-open this issue?

**Comment by chingor13** [comment]
I think we reverted the fix because of scope pollution between requests (not reset). The trigger info is not being logged on all the log messages during the request

**Comment by chingor13** [comment]
Fixed in #3516 and #3582
